package/package.py

```diff
@@ -312,6 +319,21 @@ def build_package_archive(metadata: BuildMetadata, package_dir: str) -> str:
     return archive_path
 
 
+def build_debug_package_archive(metadata: BuildMetadata, package_dir: str, mountpoint_artifact: MountpointArtifact) -> str:
+    """Build a debug.tar.gz archive from the contents of the package directory including the debug information.
+    Return the path to the final debug.tar.gz archive."""
+
+    archive_path = os.path.join(metadata.output_dir, metadata.artifact_name("debug.tar.gz"))
+
+    with tempfile.TemporaryDirectory() as tmpdir:
```

Comment:
I'd recommend using the built in [`tarfile`](https://docs.python.org/3/library/tarfile.html) library for this rather than shelling out. Alternatively [`shutil.make_archive`](https://docs.python.org/3/library/shutil.html#shutil.make_archive) exists.

Reply:
Same with other comment, I agree but the whole code uses `run` to shell out to do things, I'd rather to have a separate PR to address it.